superset/assets/src/dashboard/components/filterscope/FilterScopeSelector.jsx

```diff
@@ -304,17 +299,28 @@ export default class FilterScopeSelector extends React.PureComponent {
   onSave() {
     const { filterScopeMap } = this.state;
 
-    console.log(
-      'i am current state',
-      this.allfilterFields.reduce(
-        (map, key) => ({
+    const allFilterFieldScopes = this.allfilterFields.reduce(
+      (map, filterKey) => {
+        const { chartId } = getChartIdAndColumnFromFilterKey(filterKey);
+        const nodes = filterScopeMap[filterKey].nodes;
+        // remove filter's chart id from its own scope
+        const checkedChartIds =
+          filterScopeMap[filterKey].checked.filter(id => id !== chartId) || [];
+
+        return {
           ...map,
-          [key]: filterScopeMap[key].checked,
-        }),
-        {},
-      ),
+          [filterKey]: getFilterScopeFromNodesTree({
+            filterKey,
+            nodes,
+            checkedChartIds,
+          }),
+        };
+      },
+      {},
     );
 
+    this.props.updateDashboardFiltersScope(allFilterFieldScopes);
+    this.props.setUnsavedChanges(true);
```

Comment:
What does passing `true` here mean? seems like `setUnsavedChanges` wouldn't require an argument

Reply:
```
export function setUnsavedChanges(hasUnsavedChanges) {
  return { type: SET_UNSAVED_CHANGES, payload: { hasUnsavedChanges } };
}
```
i need to notify dashboard state that there is unsaved changes, since dashboardFilters state is updated.